### ISSUE TITLE
Filter cards by date

### ISSUE THREAD

**Issue by ebxm** [wekan/wekan#935]
Hello, 

We would need a filter by date on wekan. I haven't seen such a feature on the latest version.

It could allow for example to view cards deplaced at a specific date, or to view created cards between two dates, etc.

Is it something that would interest others ? 

Thanks in advance.

**Comment by neooleg** [comment]
Hi guys,

I can define requirements (RFC) for the feature, if somebody will be able to take it in development?

**Comment by xet7** [comment]
@neooleg 

Yes, please add all details to issues what is required, how it should work etc.

**Comment by neooleg** [comment]
OK, there is draft v 0.1 with some TODOs (TBA) for later work, but basic idea should be clear.

## General

- Filtering should be **very simple for user** (i.e. _without_ any queries, operators, operands, etc.) and **extendable on implementation level**. Let’s call it _Filtering Engine_.
- All existing features and all future plugin API should be covered with _Filtering Engine_, i.e. the filtering architecture should allow to implement filtering for any existing field and all added fields by plugins (see Technical Details below).
- There are only _two types_ of filtering for user:
  - Text filtering
  - Selector (checkbox or radio button) filtering.
- Clear filter button should be the last one on filter panel.

## Text Filtering

- On filter panel should be single _text box_ which accepting any text
- Typed text should **immediately** show only cards which contains this text in _any_ of the following fields:
  - Title
  - Description
  - Text within checklists
  - Comments

## Selector Filtering

The following fields should be represented as _sections_ within a _set of selectors inside_:
- Label (checkbox type)
  - List of labels on the board
- Due (radio button type)
  - Due in the next day
  - Due in the next week
  - Due in the next month
  - Has no due date
- Last updated (radio button type)
  - Updated a day ago
  - Updated a week ago
  - Updated a month ago
  - Updated more than a month ago
- Created at (radio button type)
  - Created a day ago
  - Created a week ago
  - Created a month ago
  - Created more than a month ago
- Created by (checkbox type)
  - List of members added to the board
- Assigned to (checkbox type)
  - List of members added to the board

## Technical Details

### Idea
- The general idea to have implementation which allow **to compose** filter logic with simple picking a field as operand and combining them with operators. It's logic of the filter.
- The logic should be able to expose on UI. It's view of the filter.

_**TBA**_

## Wireframes

_**TBA**_

**Comment by ebxm** [comment]
This is great it perfectly comply with our ideas.

Thank you !

**Comment by ebxm** [comment]
We have another need linked to the filter idea is sorting.

ex : 
* Sorting by tags (we use priority colors : P1 is red, P2 is orange, P3 is yellow and would like a sort by priority) ;
* Sorting by dates (beginning, deadline, etc.) ;

It could be an option of a column or a global option like filter. 

Does it need a specific issue or can it be part of this one ?

Thanks

**Comment by neooleg** [comment]
@ebxm, this is a separate feature, so should be a separate issue IMO.

**Comment by Creanimo** [comment]
This would be very useful!
May I add:
- due in the previous week (great for weeklies)
- overdue

How about a way to save presets of combined filters like "all of my cards due in previous week"?

**Comment by mpathiuk** [comment]
I'd also like to be able to filter by overdue cards. I'm not sure if the advanced filter allows operands on the 'due date' field currently, I've tried but with no luck so far.

**Comment by GusevVictor** [comment]
Can I show due - overdue cards..? I'm not shure how to do it... Is it possible at all?

**Comment by xet7** [comment]
@GusevVictor 

Filtering by due date etc dates is not implemented yet.

There is only filtering by Custom Fields (Text/Number/Date/DropDown List) so if you add some Date Custom Field to cards, then you can filter those, using Advanced Filter.

**Comment by xet7** [comment]
@GusevVictor 

https://github.com/wekan/wekan/wiki/Custom-Fields

**Comment by xet7** [comment]
In newest Wekan release that was released today, also Custom Fields can have markdown.

**Comment by GusevVictor** [comment]
> 
> 
> @GusevVictor
> 
> Filtering by due date etc dates is not implemented yet.
> 
> There is only filtering by Custom Fields (Text/Number/Date/DropDown List) so if you add some Date Custom Field to cards, then you can filter those, using Advanced Filter.

Thank a lot for you job. It is very important to have a native filter with due date I think. For example: I have about 10 project. I want to open wekan and show any 'red' due date tasks from any project.

**Comment by TheBigBloodyB** [comment]
Hi there all,

is there any way to use the filter in the calendar also?

My intention is to NOT see cards with a Tag or custom field "done" (or like this) in the Calendar List. 
At the moment i have done cards from the past in this view:

![image](https://user-images.githubusercontent.com/74789895/99835525-06296700-2b65-11eb-8e2f-2629e22c99c6.png)

**Comment by xet7** [comment]
@TheBigBloodyB 

Please add separate new issue.

**Comment by avatar1024** [comment]
> How about a way to save presets of combined filters like "all of my urgent cards due in next week"?

This would actually be super useful! Should we open a new feature request for this?
